src/Data/Markdown/HtmlWhiteLists.hs

```diff
@@ -0,0 +1,982 @@
+        HtmlElements <$> (withText "element" (pure . mk) `mapM` els1)
+
+
+-- | every attribute consists of a maybe-element and an attribute.  if the element is nothing, the
+-- attribute is allowed in all elements.
+newtype HtmlAttributes = HtmlAttributes [(Maybe (CI ST), CI ST)]
+  deriving (Eq, Ord, Show, Read, Monoid)
+
+instance FromJSON HtmlAttributes where
+    parseJSON v = do
+        els0 :: Value   <- withObject "html attribute whitelist" (.: "allowed") v
+        els1 :: [Value] <- withArray "element list" (pure . Vector.toList) els0
+
+        let parseAttr s = o s <|> p s
+
+            o = withObject "key with comment" (.: "key") >=> p
+            p = withText "attr" $ \case (ST.splitOn "::" -> [el, attr]) -> pure (mk <$> f el, mk attr)
+                                        bad -> fail ("no element constraint: " <> show bad)
+            f "*" = Nothing
+            f el  = Just el
+
+        HtmlAttributes <$> parseAttr `mapM` els1
+
+newtype Css3Properties = Css3Properties [CI ST]
+  deriving (Eq, Ord, Show, Read)
+
+instance FromJSON Css3Properties where
+    parseJSON v = do
+        els0 :: Value   <- withObject "css elem whitelist" (.: "allowed") v
+        els1 :: [Value] <- withArray "element list" (pure . Vector.toList) els0
+        Css3Properties <$> (withText "element" (pure . mk) `mapM` els1)
+
+
+htmlElements :: HtmlElements
+htmlElements = html4Elements <> html5Elements
+
+-- /src/com/google/caja/lang/html/html4-elements-whitelist.json
+html4Elements :: HtmlElements
+html4Elements = unsafeFromJSON [aesonQQ|
+{
+  "description": [
+      "See http://code.google.com/p/google-caja/wiki/CajaWhitelists",
+      "The denied is not necessary but lets us document why they're denied."
+      ],
+
+  "allowed": [
+      "A",
+      "ABBR",
+      "ACRONYM",
+      "ADDRESS",
+      "AREA",
+      "B",
+      "BDO",
+      "BIG",
+      "BLOCKQUOTE",
+      "BR",
+      "BUTTON",
+      "CAPTION",
+      "CENTER",
+      "CITE",
+      "CODE",
+      "COL",
+      "COLGROUP",
+      "DD",
+      "DEL",
+      "DFN",
+      "DIR",
+      "DIV",
+      "DL",
+      "DT",
+      "EM",
+      "FIELDSET",
+      "FONT",
+      "FORM",
+      "H1",
+      "H2",
+      "H3",
+      "H4",
+      "H5",
+      "H6",
+      "HR",
+      "I",
+      "IFRAME",
+      "IMG",
+      "INPUT",
+      "INS",
+      "KBD",
+      "LABEL",
+      "LEGEND",
+      "LI",
+      "MAP",
+      "MENU",
+      "OL",
+      "OPTGROUP",
+      "OPTION",
+      "P",
+      "PRE",
+      "Q",
+      "S",
+      "SAMP",
+      "SELECT",
+      "SMALL",
+      "SPAN",
+      "STRIKE",
+      "STRONG",
+      "SUB",
+      "SUP",
+      "TABLE",
+      "TBODY",
+      "TD",
+      "TEXTAREA",
+      "TFOOT",
+      "TH",
+      "THEAD",
+      "TR",
+      "TT",
+      "U",
+      "UL",
+      "VAR"
+      ],
+
+  "denied": [
+      { "key": "APPLET",
+        "reason": "disallow because allows scripting" },
```

Comment:
yes i assume so.  this is just copied verbatim from caja, and i'm only using the white lists.  as i indicate in the haddock of `Markdown` though, it is possible that there are some subtleties in the semantics of these json files that i've missed.